pkgs/games/unciv/default.nix

```diff
@@ -0,0 +1,57 @@
+{ stdenv
+, fetchurl
+, makeDesktopItem
+, makeWrapper
+, jre
+, libpulseaudio
+, libXxf86vm
+}:
+let
+  desktopItem = makeDesktopItem {
+    name = "unciv";
+    exec = "unciv";
+    comment = "Unciv, an open-source Android/Desktop remake of Civ V";
+    desktopName = "Unciv";
+    categories = "Game;";
+  };
+
+  envLibPath = stdenv.lib.makeLibraryPath [
+    libpulseaudio
+    libXxf86vm
+  ];
+
+in
+stdenv.mkDerivation rec {
+  pname = "unciv";
+
+  version = "3.9.4";
+
+  src = fetchurl {
+    url = "https://github.com/yairm210/Unciv/releases/download/${version}/Unciv.jar";
+    sha256 = "0i78mb0vby5v4hrdggfqp4d5iznixygm92y8d6hcgrf777nibb73";
+  };
+
+  phases = "installPhase";
+
+  buildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    mkdir -p $out/opt
+    cp ${src} $out/opt/Unciv.jar
+
+    ${desktopItem.buildCommand}
+
+    makeWrapper ${jre}/bin/java $out/bin/unciv \
+      --prefix LD_LIBRARY_PATH : ${envLibPath} \
+      --prefix PATH : ${stdenv.lib.makeBinPath [ jre ]} \
```

Comment:
Why is this needed? Doesn't `jre` aware of itself?

Reply:
Well, I do not know how else to create unciv script (link) that would call jre with the given Unciv.jar.

Reply:
If I get you right, you have a slight misunderstanding. Your current `makeWrapper` call generates a script at `$out/bin/unciv` that calls `${jre}/bin/java` with the env vars you specified.

Reply:
And that is exactly what I want. User executes `unciv` and it start java with proper parameters including -jar parameter pointing to jar archive with Unciv.